facebook/react
# Attach scroll/wheel to root: alternative scroll browser locking fix

This is an alternative solution to https://github.com/facebook/react/pull/9333 that attaches scroll/wheel listeners to the root container element.

So far, this prevents browser scroll locking in Chrome, Edge and Firefox. **It does not appear to fix locking on Safari**. However I think I might be getting inaccurate readings in BrowserStack and would really appreciate some help testing.

### Test Plan

1. Visit http://nh-react-scroll-root.surge.sh/scroll
2. Scroll around using the window for 10 seconds
3. Scroll around using the inner scrolling div for 10 seconds

**Expected:** No scroll locking

**Browsers**

- [x] Chrome 69
- [x] Edge 17
- [x] Firefox 61
- [ ] Safari 11.1
